### PR TITLE
2127 flat, dark, background in pyFAI-calib2 [PROPOSAL]

### PR DESCRIPTION
**Pull request by EdgarGF93**
Proposal for including dark current and flat field images in pyFAI-calib2

- [x] Dark current file(s)
- [x] Flat field file

Idea:
- Currently, pyFAI-calib2 works with ImageModels to handle the data arrays from the file and the mask. **ImageFromFilenameModel** is a generic class that handles the filename and the data array. **MaskedImageModel** computes the masking of the array.
- Idea: to include two new classes:
1. **DarkedMaskedImageModel**, same as MaskedImageModel but also takes an instance of ImageFromFilenameModel for the dark current file, and computes the masking and following subtraction of the dark current array

darked_masked_data = Masking(data_array) - dark_current_array

2. **CorrectedImageModel**, takes two instances of DarkedMaskedImageModel and an instance of ImageFromFilenameModel for the flat field file, and computes the subtraction of both DarkedMaskedImageModel following by a normalization with flat field array. The idea is to use the second instance of DarkedMaskedImageModel as a background file. This is the correction protocol:

corrected_image = (darked_masked_image_data - darked_masked_background_data) / flat_field

inspired by: https://lookingatnothing.com/index.php/archives/2251

### CODE REVIEW

File: src/pyFAI/gui/model/DarkedMaskedImageModel.py
Comment (kif) [inline src/pyFAI/gui/model/DarkedMaskedImageModel.py:1]: Can't this be replaced by preprocessedImageModel ?

Comment (EdgarGF93) [reply src/pyFAI/gui/model/DarkedMaskedImageModel.py:1]: correct, DarkedMaskedImageModel.py is no longer needed

**Comment by kif** [comment]
Is this still WIP ?

**Comment by EdgarGF93** [comment]
Yes, I will test the change of the image and I will label Ready to merge

**Comment by EdgarGF93** [comment]
now, flat and dark images can be uploaded through CLI, dialog file and drag&drop, and they will be properly computed along all the calib2 tasks

**Comment by kif** [comment]
@jonwright are you willing to test it ?

**Review by kif — APPROVED** [review]
LGTM

**Comment by jonwright** [comment]
Sure - we will try this later today or during the weekend - sometime soon after we figure out how to install pyFAI from a pull request.

**Comment by kif** [comment]
you can use "./bootstrap pyFAI-calib2" to test locally without installing. 
You may install all dependencies before-hand with a a "pip install -r requirements.txt --upgrade"

**Pull request merged**